spec/tree_spec.rb

```diff
@@ -50,6 +50,37 @@
     it "has blobs" do
       expect(@tree.blobs).to be_kind_of Array
     end
+
+    it "finds blobs and trees efficiently" do
+      count = 0
+      result = @bare_repo.head.tree.find do |tree_entry|
+        count = count+1
+        tree_entry[:name] == '.gitignore'
+      end
+      expect(result).to be_a Blob
+      expect(result.id).to eq 'baaa47163a922b716898936f4ab032db4e08ae8a'
+      expect(count).to eq 1 # .gitignore is first in the tree, so no more than one iteration should have been performed.
```

Comment:
Should this count be verified by checking another blob as well?

Reply:
I've changed the spec to use the third file, which tests the logic clearly enough I think.